### PR TITLE
Transforms blog

### PR DESCRIPTION
**Pull request by justinvp**
Add a blog post on the transforms APIs. Originally ported from https://github.com/pulumi/pulumi-hugo/pull/4266.

## Tasks
- [x] https://github.com/pulumi/docs/pull/11851
- [x] https://github.com/pulumi/pulumi-dotnet/pull/298
- [x] Release a new version of https://github.com/pulumi/pulumi-dotnet w/ above fix ([v3.65.0](https://github.com/pulumi/pulumi-dotnet/releases/tag/v3.65.0))
- [x] Add meta image

### CODE REVIEW

File: content/blog/announcing-new-transform-callbacks/index.md [hunk 1]
```diff
@@ -0,0 +1,23 @@
+---
+title: "Announcing new transform callbacks"
+date: 2024-05-15
+meta_desc: Transform callbacks are a rebuild of the transformations system to work for Multi-Language Components
+meta_image: meta.png
```
Comment (sean1588) [inline content/blog/announcing-new-transform-callbacks/index.md:5]: FYI - this is likely failing because the meta.png `meta_image` does not yet exist.

**Comment by EronWright** [comment]
Related: https://github.com/pulumi/docs/pull/11851

**Review by Frassle — APPROVED** [review]

**Review by Frassle — COMMENTED** [review]
Need a new meta image, but text content looks ok to me

File: content/blog/resource-transforms/index.md [hunk 2]
```diff
@@ -0,0 +1,168 @@
+---
+title: "Resource Transforms"
```
Comment (meagancojocar) [inline content/blog/resource-transforms/index.md:2]:
```suggestion
title: "Introducing: Resource Transforms in the Pulumi CLI"
```

File: content/blog/resource-transforms/index.md [hunk 3]
```diff
@@ -0,0 +1,168 @@
...
+date: 2024-07-16
```
Comment (justinvp) [inline content/blog/resource-transforms/index.md:3]:
```suggestion
date: 2024-07-18
```

**Pull request merged**